rust/src/types.rs

```diff
@@ -1,29 +1,99 @@
-use std::sync::Arc;
-use bdk::blockchain::AnyBlockchain;
-use crate::ffi::{Balance, Wallet};
+
 use serde::{Deserialize, Serialize};
 
-#[derive(Clone)]
-pub(crate) struct  BdkInfo{
-    pub wallet: Option<Arc<Wallet>>,
-    pub blockchain: Option<Arc<AnyBlockchain>>,
+pub struct TxOut {
+    /// The value of the output, in satoshis.
+    pub(crate) value: u64,
+    /// The address of the output.
+    pub(crate) address: String,
+}
+#[derive(PartialEq)]
+pub enum DescriptorKeyType{
+    EXTENDED,
+    DERIVED,
+    DEFAULT
+}
```

Comment:
What does this type do? I see it used in the DescriptorSecretKey type but I'm not sure I fully understand its purpose.